### PR TITLE
Close 41 create user scenarios for running jenkins

### PR DESCRIPTION
**Pull request by buep**
Proposals for some usage scenarios we can use to detect more missing features, and that later can be used as documentation helping users migrate.

### CODE REVIEW

File: docs/usageScenarios.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Their Jenkins global configuration is the only piece not under version control, so they depend on the current data in `JENKINS_HOME` to persist the Jenkins global configuration and be able to redeploy the Jenkins master with existing installed plugins. They do not care about their historic build data.
+
+**With Jenkins configuration as code plugin** they can get their missing configuration under version control as well and their installed plugin combinations and for this little full stack developer team it means they can always just redeploy their Jenkins should something be wrong or failing, and their container orchestration tool will keep their Jenkins master always running for them completely automatically.
+
+**Migrating to use the Jenkins Configuration as Code plugin** is easy if they can live with doing just few manual steps before it is all automated. They need to install the Jenkins Configuration as Code plugin on their current master, then use the export functionality to create the `plugins.yml` and `jenkins.yml` file for them, which they need to add to a git repository.
```
Comment (ewelinawilkosz) [inline docs/usageScenarios.md]: I'm wondering if maybe we could also add here a case where we use plugins.txt to install the plugins - including our plugin and feed docker with jenkins.yaml location in an environment variable plugin recognises? is the same case you're describing or separate one?

Comment (buep) [reply docs/usageScenarios.md]: Yes the same case, but couldn't mention the docker example command as this team uses an orchestration tool so they describe this in a deployment file...  but conceptually it is the same setup.

Comment (buep) [reply docs/usageScenarios.md]: Updated the description to elaborate the difference between plugins.txt and plugins.yml and why we need a image to support their initial setup.

As part of this issue by the way I wrote some issues on things we need: #65 and now #70

File: docs/usageScenarios.md [hunk 2]
```diff
@@ -48,19 +48,24 @@ _Migration step-by-step guide:_ For the exact steps, see this part of the migrat
...
+Then the last thing they need to do is to update their deployment recipe to use the customized Jenkins where the Jenkins Configuration as Code plug-in comes pre-installed, and have the recipe pass the URI to the `jenkins.yml` file when starting Jenkins.
 
 After that they only do changes in the configuration file through git, and Jenkins Configuration as Code will refresh global configuration. Should their infrastructure fail their orchestration tool will recreate their master from scratch, but still with the same configuration and without depending on old data sets.
 
+Notice a slight difference between `plugins.yml` as part of the Configuration as Code vs `Plugins.txt` known from the Jenkins container setups. The latter one will need to contain all plugins and their dependencies and will usually be auto generated to support starting a Jenkins container with all the plug-ins one like. The `plugin.yml` takes the approach of only mentioning the plugin and their version (or latest) the user really cares about, and Configuration as Code will resolve dependencies for the user.
```
Comment (ewelinawilkosz) [inline docs/usageScenarios.md]: I now have only problem with capital 'P' in Plugins.txt :)

Comment (buep) [reply docs/usageScenarios.md]: fixed

**Pull request merged**